File: PECGI_SPC/Class/clsProcess.vb
Imports System.Data.SqlClient

Public Class clsProcess
    Public Property FactoryCode As String
    Public Property ProcessCode As String
    Public Property ProcessName As String
    Public Property LineGroup As String
    Public Property ProcessGroup As String
End Class

Public Class clsProcessDB
    Public Shared Function GetList(UserID As String, FactoryCode As String, ProcessGroup As String, LineGroup As String, Optional ShowAll As Boolean = False) As List(Of clsProcess)
        Using Cn As New SqlConnection(Sconn.Stringkoneksi)
            Cn.Open()
            Dim cmd As New SqlCommand("sp_SPC_FillCombo", Cn)
            cmd.CommandType = CommandType.StoredProcedure
            cmd.Parameters.AddWithValue("ComboType", 2)
            cmd.Parameters.AddWithValue("UserID", UserID)
            cmd.Parameters.AddWithValue("FactoryCode", FactoryCode)
            cmd.Parameters.AddWithValue("ProcessGroup", ProcessGroup)
            cmd.Parameters.AddWithValue("LineGroup", LineGroup)
            If ShowAll Then
                cmd.Parameters.AddWithValue("All", 1)
            End If
            Dim rd As SqlDataReader = cmd.ExecuteReader
            Dim ProcessList As New List(Of clsProcess)
            Do While rd.Read
                Dim Process As New clsProcess
                Process.FactoryCode = rd("FactoryCode")
                Process.ProcessCode = rd("ProcessCode")
                Process.ProcessName = rd("ProcessName")
                Process.LineGroup = rd("LineGroup")
                Process.ProcessGroup = rd("ProcessGroup")
                ProcessList.Add(Process)
            Loop
            rd.Close()
            Return ProcessList
        End Using
    End Function
    Public Shared Function GetListForItemCheckByType(UserID As String, FactoryCode As String, ProcessGroup As String, LineGroup As String, Optional ShowAll As Boolean = False) As List(Of clsProcess)
        Using Cn As New SqlConnection(Sconn.Stringkoneksi)
            Cn.Open()
            Dim cmd As New SqlCommand("sp_SPC_FillCombo", Cn)
            cmd.CommandType = CommandType.StoredProcedure
            cmd.Parameters.AddWithValue("ComboType", 4)
            cmd.Parameters.AddWithValue("UserID", UserID)
            cmd.Parameters.AddWithValue("FactoryCode", FactoryCode)
            cmd.Parameters.AddWithValue("ProcessGroup", ProcessGroup)
            cmd.Parameters.AddWithValue("LineGroup", LineGroup)
            If ShowAll Then
                cmd.Parameters.AddWithValue("All", 1)
            End If
            Dim rd As SqlDataReader = cmd.ExecuteReader
            Dim ProcessList As New List(Of clsProcess)
            Do While rd.Read
                Dim Process As New clsProcess
                Process.FactoryCode = rd("FactoryCode")
                Process.ProcessCode = rd("ProcessCode")
                Process.ProcessName = rd("ProcessName")
                Process.LineGroup = rd("LineGroup")
                Process.ProcessGroup = rd("ProcessGroup")
                ProcessList.Add(Process)
            Loop
            rd.Close()
            Return ProcessList
        End Using
    End Function
End Class

File: SPCMeasurement/SPCMeasurement/clsProcess.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Data;
using System.Data.SqlClient;
using C1.Win.C1List;

namespace SPCMeasurement
{
    class clsProcess
    {
    }

    class clsProcessDB
    {
        private static clsConfig cfg = new clsConfig();
        private static clsConfigData cfd = cfg.uf_ReadConfig();
        private static string constr = cfd.ConnectionString;

        public static void FillCombo(C1.Win.C1List.C1Combo cbo, string UserID, string FactoryCode, string ProcessGroup, string LineGroup)
        {

            cbo.DataMode = DataModeEnum.Normal;
            using (SqlConnection cn = new SqlConnection(constr))
            {
                cn.Open();
                string q = "sp_SPC_FillCombo";
                SqlCommand cmd = new SqlCommand(q, cn);
                cmd.CommandType = CommandType.StoredProcedure;
                cmd.Parameters.AddWithValue("ComboType", 2);
                cmd.Parameters.AddWithValue("UserID", UserID);
                cmd.Parameters.AddWithValue("FactoryCode", FactoryCode);
                cmd.Parameters.AddWithValue("ProcessGroup", ProcessGroup);
                cmd.Parameters.AddWithValue("LineGroup", LineGroup);
                SqlDataAdapter da = new SqlDataAdapter(cmd);
                DataTable dt = new DataTable();
                da.Fill(dt);
                cbo.DataSource = dt;
                cbo.ColumnHeaders = false;
                cbo.DataMode = DataModeEnum.Normal;
                cbo.ItemHeight = 18;
                cbo.MaxDropDownItems = 10;
                cbo.Splits[0].DisplayColumns[0].Visible = false;
                cbo.Splits[0].DisplayColumns[1].Visible = false;
                cbo.Splits[0].DisplayColumns[2].Visible = false;
                cbo.Splits[0].DisplayColumns[3].Visible = false;
                cbo.Splits[0].DisplayColumns[4].Visible = false;
                cbo.ExtendRightColumn = true;
                cbo.DropdownPosition = C1.Win.C1List.DropdownPositionEnum.LeftDown;
                cbo.ValueMember = "ProcessCode";
                cbo.DisplayMember = "ProcessName";
                cbo.LimitToList = true;
            }
        }
    }
}
